lib/AST/ASTContext.cpp

```diff
@@ -2912,6 +2912,18 @@ Type ErrorType::get(Type originalType) {
   return entry = new (mem) ErrorType(ctx, originalType, properties);
 }
 
+Type UnresolvedType::get(Type originalType) {
+  assert(originalType);
```

Comment:
I think it would be reasonable to assert here that original type is not a `TypeVariableType`

Reply:
Or better yet !type->hasTypeVariableType()

Reply:
Yeah, sorry that's what I meant :)